### ISSUE TITLE
Counting subgraphs in sources is slow

### ISSUE THREAD

**Issue by noah-aviel-dove** [DataBiosphere/azul#3644]
We have to list subgraphs in a source in order to determine their count.

**Comment by noah-aviel-dove** [comment]
We should add a `count_bundles(source, prefix)` method to the repository plugin. A default implementation would just fall back to `list_bundles`. The TDR Implementation would use  `select count() from links`.

**Comment by noah-aviel-dove** [comment]
Might be superseded by work on https://github.com/DataBiosphere/azul/pull/3583

**Comment by melainalegaspi** [comment]
@hannes-ucsc : "Will leave this in triage until we are certain that this ticket is superseded by #3583, most likely when that PR is merged."

**Comment by hannes-ucsc** [comment]
Blocker #3583 has been approved. Spike to determine whether is ticket is obsolete or still valid.

**Comment by noah-aviel-dove** [comment]
The `list_partitions` method added in `https://github.com/DataBiosphere/azul/pull/3583` isn't meaningfully faster than using listing the bundles and counting them. However I can't find any call to `*.list_bundles` where we only use the bundles' count and not the actual FQIDs. Therefore, if we were to add the `count_bundles` method proposed in my earlier comment, I don't think we could actually use it anywhere.

In summary, this issue was not superseded, but I think it's invalid anyway.
